optuna/study.py

```diff
@@ -7,22 +7,25 @@
 import pandas as pd
 from six.moves import queue
 import time
-from typing import Any  # NOQA
-from typing import Callable  # NOQA
-from typing import Dict  # NOQA
-from typing import List  # NOQA
-from typing import Optional  # NOQA
-from typing import Set  # NOQA
-from typing import Tuple  # NOQA
-from typing import Type  # NOQA
-from typing import Union  # NOQA
+from typing import Callable
```

Comment:
Can we move this import statement `Callable` and the definition of `ObjectiveFuncType` to the `if types.TYPE_CHECKING` block? `Callable` is used to define `ObjectiveFuncType` at L30, and `ObjectiveFuncType` is only used for type checking. So, I think they are not required to execute code.

Reply:
Yes, we can. Thank you for pointing it out.

Reply:
Fixed: https://github.com/pfnet/optuna/pull/326/commits/ec275748e7bf404e37f5e33dea542e030d6b7ff4